### ISSUE TITLE
Edge validity is not checked when using named graphs

### ISSUE THREAD

**Issue by uditha-atukorala** [arangodb/arangodb#2050]
It seems like when inserting into edge collections through named graphs edge validity is not checked. 

```
$ curl -X POST --data-binary @- --dump - http://localhost:8529/_api/database <<EOF
> { "name" : "example" }
> EOF
HTTP/1.1 201 Created
Content-Type: application/json; charset=utf-8
Server: ArangoDB
Connection: Keep-Alive
Content-Length: 40

{"result":true,"error":false,"code":201}
```

```
$ curl -X POST --data-binary @- --dump - http://localhost:8529/_db/example/_api/gharial <<EOF
> {
>   "name" : "acl",
>   "edgeDefinitions" : [
>     {
>       "collection": "belongs_to",
>       "from": [
>         "acl"
>       ],
>       "to": [
>         "acl"
>       ]
>     }
>   ]
> }
> EOF
HTTP/1.1 202 Accepted
Etag: 1649
Content-Type: application/json; charset=utf-8
Server: ArangoDB
Connection: Keep-Alive
Content-Length: 182

{"error":false,"graph":{"name":"acl","edgeDefinitions":[{"collection":"belongs_to","from":["acl"],"to":["acl"]}],"orphanCollections":[],"_id":"_graphs/acl","_rev":"1649"}
```

```
$ curl -X POST --data-binary @- --dump - http://localhost:8529/_db/example/_api/gharial/acl/edge/belongs_to <<EOF
> {
>   "_from" : "acl/dummy-1",
>   "_to" : "acl/dummy-2"
> }
> EOF
HTTP/1.1 202 Accepted
Etag: 1744
Content-Type: application/json; charset=utf-8
Server: ArangoDB
Connection: Keep-Alive
Content-Length: 87

{"error":false,"edge":{"_id":"belongs_to/1744","_key":"1744","_rev":"1744"},"code":202}
```

With the above, I shouldn't have been able to insert `belongs_to/1744` edge since `acl/dummy-1` and `acl/dummy-2` vertices don't exist.
## my environment running ArangoDB

I'm using the latest (v3.0.7) ArangoDB of the respective release series:
- [ ] 2.8
- [x] 3.0
- [ ] self-compiled devel branch

On this operating system:
- [ ] DCOS on
  - [ ] AWS
  - [ ] Azure
  - [ ] own infrastructure
- [x] Linux 
  - [ ] Debian .deb
  - [ ] Ubuntu .deb
  - [ ] SUSE   .rpm
  - [ ] RedHat .rpm
  - [ ] Fedora .rpm
  - [ ] Gentoo
  - [x] docker - official docker library
  - [ ] other:
- [ ] Windows, version: 
- [ ] MacOS, version:

**Comment by uditha-atukorala** [comment]
Any update on this?

**Comment by weinberger** [comment]
Sorry for the late response. It's a bug and we will have a look.

**Comment by saliormoon** [comment]
I have the same problem

**Comment by hkernbach** [comment]
This is fixed in 3.3 and upcoming versions.
